platform/android/llapp_main.lua

```diff
@@ -16,53 +16,11 @@ end
 -- path to primary external storage partition
 local path = android.getExternalStoragePath()
 
--- run user shell scripts or recursive migration of user data
-local function runUserScripts(dir, migration, parent)
-    for entry in lfs.dir(dir) do
-        if entry ~= "." and entry ~= ".." then
-            local fullpath = dir .. "/" .. entry
-            local mode = lfs.attributes(fullpath).mode
-            if mode == "file" and migration then
-                if entry ~= "migrate" and not fullpath:match(".sh$") then
-                    local destdir = parent and android.dir .. "/" .. parent or android.dir
-                    -- we cannot create new directories on asset storage.
-                    -- trying to do that crashes the VM with error=13, Permission Denied
-                    android.execute("cp", fullpath, destdir .."/".. entry)
-                end
-            elseif mode == "file" and fullpath:match(".sh$") then
-                android.execute("sh", fullpath, path .. "/koreader", android.dir)
-            elseif mode == "directory" then
-                runUserScripts(fullpath, migration, parent and parent .. "/" .. entry or entry) -- recurse into next directory
-            end
-        end
-    end
-end
-
+--[[-- commented out if favor or `livepatch.lua` (#9104)
```

Comment:
Please remove the comments.

Also: which instructions an user of the now legacy "patch.lua"must follow to migrate their patches to the new system?

If that's too wild I would suggest to leave the patch.lua option in here (guarded by android.prop.runtimeChanges) and we're going to be sure we don't break the setups of android users with crappy ereaders :)

Reply:
I the order of your comment:

1.) I will remove the addressed comments as you like (then `llapp_main.lua` will be virgin again  ;-) )

2.) I designed this whole thing, as a drop in replacement for my old script hack and the Android specific `patch.lua`. The goal is, to apply patches (early_once, early and so on) on some defined runtime states of KOReader. (Not all of them are used by now, but IMHO this is future proof.)

I would expect that `patch.lua` gets renamed to `userpatches/1-patch.lua` and be executed in `reader.lua` quite after start (but after "userpach_early" in line 17 `require("setupkoenv")`. So this should be safe – except a `patch.lua` is messing with Lua's or KOReader's path's.
So the only instruction for existing patches would be: Do nothing, let the magic happen.
For new patches the instruction would be: "Put your changes **not** in `DataStorage:getDataDir()\patch.lua`! Put it in `DataStorage:getDataDir()\userpatches\1-patch.lua`. You can have multiple patches numbered with `xxx` in `1xxx-patches.lua`."
The execution order of the patches is natural(!). So 1-patch is the first continuing with 10.patch to  19-patch then comes 100-patch to 199-patch followed by  1000-patch to 1999-patch and so on.

Although I have not seen any `patch.lua` until now, I would expect, there is nothing to do. 

Anyway, I don't own any Android E-Reader anymore, so I may be of limited help (despite, what I have learned in the past).

Reply:
Thanks for the detail!

Could I make another suggestion?

replace the dir name `userpatches` with `patch`. Because:

1. since we're in dataDir and all app data there belongs to the user ;).
2. an user wants to patch, not to userpatch :)
3. makes easier to remember: to patch create folder `patch` and put patches there.
4. A bit less to type :P